pkg/apis/serving/v1alpha1/revision_validation_test.go

```diff
@@ -198,6 +199,62 @@ func TestConcurrencyModelValidation(t *testing.T) {
 	}
 }
 
+func TestContainerConcurrencyValidation(t *testing.T) {
+	tests := []struct {
+		name string
+		cc   RevisionContainerConcurrencyType
+		cm   RevisionRequestConcurrencyModelType
+		want *apis.FieldError
+	}{{
+		name: "single with only container concurrency",
+		cc:   1,
+		cm:   RevisionRequestConcurrencyModelType(""),
+		want: nil,
+	}, {
+		name: "single with container currency and concurrency model",
+		cc:   1,
+		cm:   RevisionRequestConcurrencyModelSingle,
+		want: nil,
+	}, {
+		name: "multi with only container concurrency",
+		cc:   0,
+		cm:   RevisionRequestConcurrencyModelType(""),
+		want: nil,
+	}, {
+		name: "multi with container concurrency and concurrency model",
+		cc:   0,
+		cm:   RevisionRequestConcurrencyModelMulti,
+		want: nil,
+	}, {
+		name: "mismatching container concurrency (1) and concurrency model (multi)",
+		cc:   1,
+		cm:   RevisionRequestConcurrencyModelMulti,
+		want: apis.ErrMultipleOneOf("containerConcurrency", "concurrencyModel"),
+	}, {
+		name: "mismatching container concurrency (0) and concurrency model (single)",
```

Comment:
I think this occurs in nature today, so we need to either have defaults set first (with my above comment), or we need to make this acceptable.

Reply:
Done.  This should be fully backward compatible now.